src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1369,12 +1369,42 @@ private BoundExpression BindNonMethod(SimpleNameSyntax node, Symbol symbol, Diag
                         }
                         else
                         {
+                            SourceLocalSymbol sourceLocal;
+                            type = null;
+
+                            if (node.SyntaxTree == localSymbolLocation.SourceTree && (object)(sourceLocal = localSymbol as SourceLocalSymbol) != null &&
+                                sourceLocal.IdentifierToken.Parent?.Kind() == SyntaxKind.Argument)
+                            {
+                                var argument = (ArgumentSyntax)sourceLocal.IdentifierToken.Parent;
+
+                                if (argument.Identifier == sourceLocal.IdentifierToken &&
+                                    argument.Type.IsVar)
+                                {
+                                    // We are referring to an out variable which might need type inference.
+                                    // If it is in fact needs inference, it is illegal to reference it in the same argument list. 
+                                    // Technically, we could support some restricted scenarios (when it is used as another argument in the same list), 
```

Comment:
The specific place in this PR is the deletion of the paragraph starting at line 21 of outvar.md in the previous revision of that file. To align the spec with the implementation, I recommend you describe a proposed change to the followin paragraph of the existing C# specification:

> Within the scope of a local variable introduced by a local-variable-declaration, it is a compile-time error to refer to that local variable in a textual position that precedes its local-variable-declarator. If the local variable declaration is implicit (§8.5.1), it is also an error to refer to the variable within its local-variable-declarator.